src/rules/no-html.js

```diff
@@ -74,48 +68,45 @@ export default {
 	},
 
 	create(context) {
+		const { sourceCode } = context;
 		const [{ allowed, allowedIgnoreCase }] = context.options;
-		const allowedElements = new Set(
-			allowedIgnoreCase ? allowed.map(tag => tag.toLowerCase()) : allowed,
-		);
+
+		/**
+		 * Normalize a tag name based on the `allowedIgnoreCase` option.
+		 * @param {string} tagName The tag name to normalize.
+		 * @returns {string} The normalized tag name.
+		 */
+		function normalizeTagName(tagName) {
+			return allowedIgnoreCase ? tagName.toLowerCase() : tagName;
+		}
+
+		const allowedElements = new Set(allowed.map(normalizeTagName));
```

Comment:
I think the previous version might be better as this would unnecessarily create a new array when there's no need for it (when `allowedIgnoreCase` is `false`).